client-js/app/elements/labrad-registry.html

```diff
@@ -59,8 +207,213 @@
       path: {
         type: Array,
         notify: true
+      },
+      socket: {
+        type: Object,
+        notify: true,
+      },
+      selectType: {
+        type: String,
+        notify: true
       }
-    }
+    },
+    ready: function() {
+      console.log(Polymer.version);
+      this.addEventListener('iron-form-submit',this.updateKey);
+    },
+    //Helper Functions
+    selectKey: function() {
+      this.selDir = null;
+      this.selectType = 'key';
+    },
+    selectDir: function(){
+      this.selKey = null;
+      this.selectType = 'dir';
+    },
+    incrementSelector: function() {
+      console.log(this.selKey);
+      //TODO increment selected key on tab
+    },
+    repopulateList: function(resp) {
+      this.selDir=null;
+      this.splice('dirs',0,this.dirs.length);
+      this.splice('keys',0,this.keys.length);
+
+      for (var i in resp.dirs) {
+        //bw = resp.dirs.length-1-i; //run through list backwards so it orders correctly
+        //console.log(resp.dirs[i]);
+        this.push('dirs',{name: resp.dirs[i], url: this.createUrl(resp.path,resp.dirs[i])});
+      }
+      for (var j in resp.keys){ 
+        //bw = resp.keys.length-1-i; //run through list backwards so it orders correctly
+        this.push('keys',{name: resp.keys[j], value: resp.vals[j]});
+      }
+    },
+    createUrl: function(path,dir) {
```

Comment:
As we've discussed, it would be nice to avoid duplicating logic like this between the typescript code and individual polymer elements. We need to figure out how to have one scheme for modularity of the code, not have both typescript module imports and html imports. But that's clearly out of scope for this change.